UI/ChatGPT/ChatGPT/ChatGPT/Presentation/MainModel.cs

```diff
@@ -0,0 +1,82 @@
+using ChatGPT.Business;
+using ChatGPT.Extensions;
+using ChatGPT.Services;
+
+namespace ChatGPT.Presentation;
+
+public partial record MainModel
+{
+	private readonly IChatService _chatService;
+	public MainModel(IChatService chatService)
+	{
+		_chatService = chatService;
+	}
+
+	public IState<string> Prompt => State.Value(this, () => string.Empty);
+
+	public IListState<Message> Messages => ListState<Message>.Empty(this);
+
+	public IState<bool> IsMessageStream => State.Value(this, () => !OperatingSystem.IsBrowser());
+
+	public IState<bool> IsStreamEnabled => State.Value(this, () => !OperatingSystem.IsBrowser());
```

Comment:
I known it goes against my own rule to prefer to use the `Is***` prefix as much as possible, but here I think it would make sense to use `Can` and `Use` prefixes.

Also the `CanStream` will never change, the is no need to use an `IFeed`.

```suggestion
	public bool CanStream { get; } = !OperatingSystem.IsBrowser();

	public IState<bool> UseStream => State.Value(this, () => CanStream);
```

Reply:
It makes sense, thanks David. Done in next commit.